README.md

```diff
@@ -1,40 +1,274 @@
+- Some sort of bash (to run the setup scripts).
 
-The token to run the bot is taken as a command line argument. For Visual Studio, you can add the token in `Property Pages > Command Arguments`, which should look like this:
+#### Linux
 
-![](https://cdn.discordapp.com/attachments/353076704945766403/680397059068919808/unknown.png)
+To build on Linux you're going to need to have Qt5 and the **developer version** of Qt5WebSockets. You will also need `qt5-qmake`.
 
+### Setting up the project
+
+You'd have to do quite a bit of work to get the bot setup if we didn't have these two scripts:
+
+- `init.sh`
+- `generate_project_files.sh`
+
+Thanks to those, all you need to do is run them in succession and you should have all the necessary files to work with.
+
+#### `init.sh`
+
+This file makes temporary folders and files to facilitate the setup. It also makes sure you have `premake5` to generate the project files and that the submodules are initialized and cloned (helpful for those who forgot to clone with `--recurse-submodules`).
+
+On Windows it will ask you for the paths of the **x86** and **x64** versions of Qt and OpenSSL. You don't need both of the versions; you can just specify one and later on build for that specific platform.
+
+> Make sure you use forward slashes instead of the backslashes.
+
+#### `generate_project_files.sh`
+
+As the name suggests, this script generates the necessary files by using [Premake](https://premake.github.io/).
+
+You can provide an action like you do with premake, but the script is configured such that it generates file for
+
+- Visual Studio 2019 if you're on Windows.
+- `qmake` if you're on Linux.
+
+### Building
+
+This section assumes that you used the default targets and didn't provide a custom action.
+
+Once the files are generated, you should see a `sln/` directory.
+
+#### On Linux
+
+`cd` into `sln` and execute `qmake` followed by `make`.
+
+#### On Windows
+
+You should find a Visual Studio solution file (`.sln`) inside `sln`. Open that and then build the solution using Visual Studio.
+
+### Running
+
+The token to run the bot is taken as a command line argument.
+
+For Visual Studio, you can add the token here:
+
+|![Visual Studio Command Args](https://cdn.discordapp.com/attachments/353076704945766403/680397059068919808/unknown.png)|
+|:--:|
+|`Project Properties` (`Alt + Enter`) > `Configuration Properties` > `Debugging` > `Command Arguments`|
+
+## ❓ Help and FAQ
+
+If you weren't able to properly carry out the setup process, or if you just want to know more Umiko in general, considering checking these questions before opening an issue:
+
+<details>
+<summary>Windows: I think I didn't add the Qt version properly in `init.sh`</summary>
+
+If you're using Visual Studio, you need not worry; you can add those using `Qt VS Tools` (a Visual Studio extension). Simply go to `Qt Options` and then use the file explorer to add the required directories after clicking on `Add`.
```

Comment:
Why are we instructing them to use Qt VS Tools when we have premake-qt set up? Kinda defeats the whole point, dunnit?